mypy/checker.py

```diff
@@ -992,8 +992,9 @@ def get_generator_return_type(self, return_type: Type, is_coroutine: bool) -> Ty
             # AwaitableGenerator, Generator: tr is args[2].
             return return_type.args[2]
         else:
-            # Supertype of Generator (Iterator, Iterable, object): tr is any.
-            return AnyType(TypeOfAny.special_form)
+            # We have a supertype of Generator (Iterator, Iterable, object)
+            # Treat `Iterator[X]` as a shorthand for `Generator[X, Any, None]`.
```

Comment:
Why? Intuitively Any makes more sense to me here.

This affects code like this, right?
```python
x: Iterator[T]
y = yield from x
```

Reply:
This results in errors for cases like https://github.com/python/typeshed/issues/11222 / `testReturnInIterator`